specs/language/conversions.tex

```diff
@@ -0,0 +1,135 @@
+  point-integral conversion, or component-wise boolean conversion.
+  \item Zero or one qualification conversion.
+\end{enumerate}
+
+Standard conversion sequences are applied to expressions, if necessary, to
+convert it to a required destination type.
+
+\Sec{Lvalue-to-rvalue conversion}{Conv.lval}
+
+\p A glvalue of a non-function type \texttt{T} can be converted to a prvalue.
+The program is ill-formed if \texttt{T} is an incomplete type. If the glvalue
+refers to an object that is not of type \texttt{T} and is not an object of a
+type derived from \texttt{T}, the program is ill-formed. If the glvalue refers to
+an object that is uninitialized, the behavior is undefined. Otherwise the
+prvalue is of type \texttt{T}.
+
+\p If the glvalue refers to an array of type \texttt{T}, the prvalue will refer
+to a copy of the array, not memory referred to by the glvalue.
+
+\Sec{Array-to-pointer conversion}{Conv.array}
+
+\p An lvalue or rvalue of type \texttt{T[]} (bounded or unbounded), can be
+converted to a prvalue of type pointer to \texttt{T}.
+[\textit{Note: \acrshort{hlsl} does not support grammar for specifying pointer or
+reference types, however they are used in the type system and must be described
+in language rules.}]
+
+\Sec{Integral conversion}{Conv.iconv}
+
+\p A glvalue of an integer type can be converted to a cxvalue of any other
+non-enumeration integer type. A prvalue of an integer type can be converted to a
+prvalue of any other integer type.
+
+\p If the destination type is unsigned, integer conversion maintains the bit pattern
+of the source value in the destination type truncating or extending the value to
+the destination type.
+
+\p If the destination type is signed, the value is unchanged if the destination
+type can represent the source value. If the destination type cannot represent
+the source value, the result is implementation-defined.
+
+\p If the source type is \texttt{bool}, the values \texttt{true} and
+\texttt{false} are converted to one and zero respectively.
+
+\Sec{Floating point conversion}{Conv.fconv}
+
+\p A glvalue of a floating point type can be converted to a cxvalue of any other
+floating point type. A prvalue of a floating point type can be converted to a
+prvalue of any other floating point type.
+
+\p If the source value can be exactly represented in the destination type, the
+conversion produces the exact representation of the source value. If the source
+value cannot be exactly represented, the conversion to a best-approximation of
+the source value is implementation defined.
+
+\Sec{Floating point-integral conversion}{Conv.fpint}
+
+\p A glvalue of floating point type can be converted to a cxvalue of integer
+type. A prvalue of floating point type can be converted to a prvalue of
+integer type. Conversion of floating point values to integer values truncates by
+discarding the fractional value. The behavior is undefined if the truncated
```

Comment:
Is "undefined" here different to "implementation defined" on line 53?

Reply:
Yea, "undefined" and "implementation defined" are defined terms earlier in the full document:

https://github.com/microsoft/hlsl-specs/blob/main/specs/language/introduction.tex#L148

Basically "implementation defined" behavior needs to be documented by the relevant part of the implementation (compiler, runtime, ISA, DXIL, SPIR-V, etc).